src/robusta/core/sinks/pagerduty/pagerduty_sink.py

```diff
@@ -48,14 +50,112 @@ def __to_pagerduty_status_type(title: str):
         else:
             return "trigger"
 
-    def write_finding(self, finding: Finding, platform_enabled: bool):
+    @staticmethod
+    def __send_changes_to_pagerduty(self, finding: Finding, platform_enabled: bool):
         custom_details: dict = {}
+        links = []
+        if platform_enabled:
+            href = finding.get_investigate_uri(self.account_id, self.cluster_name)
+            links.append({
+                "text": "🔂 See change history in Robusta",
+                "href": href
+            })
+        else:
+            links.append({
+                "text": "🔂 Enable Robusta UI to see change history",
+                "href": "https://bit.ly/robusta-ui-pager-duty"
+            })
+
+        source = self.cluster_name
+
+        custom_details["namespace"] = finding.service.namespace
+        custom_details["resource"] = f"{finding.service.resource_type}/{finding.subject.name}"
+
+        if finding.subject.node:
+            custom_details["node"] = finding.subject.node
+
+        timestamp = finding.starts_at.astimezone().isoformat()
+
+        for enrichment in finding.enrichments:
+            for block in enrichment.blocks:
+                changes = self.__block_to_changes(block, enrichment)
```

Comment:
I think it's better to add here:
```
if not isinstance(block, KubernetesDiffBlock):
  continue
```

Then, `__block_to_changes`:
```
    @staticmethod
    def __block_to_changes(block: KubernetesDiffBlock, enrichment: Enrichment) -> Optional[Dict[str, Any]]:
        operation = enrichment.annotations.get("operation")

        change_count = 0
        if operation == K8sOperationType.UPDATE:
            change_count = block.num_modifications

        return {
            "change_count": change_count,
            "operation": operation,
        }
```

Then, no need for:
```
                if not changes:
                    continue
```
only:
```
                operation = changes["operation"]
                if not operation:  # KubernetesDiffBlock diff block without "operation" annotation
                    continue
```

WDYT?

Reply:
I wanted made it future proof and generic. I have made the changes as been suggested. Anyway in the future we could make the changes as required.